src/backend/xdg_desktop_portal.rs

```diff
@@ -7,11 +7,22 @@ use crate::message_dialog::MessageDialog;
 use crate::{FileDialog, FileHandle, MessageButtons, MessageDialogResult};
 
 use ashpd::desktop::file_chooser::{FileFilter, OpenFileRequest, SaveFileRequest};
-// TODO: convert raw_window_handle::RawWindowHandle to ashpd::WindowIdentifier
-// https://github.com/bilelmoussaoui/ashpd/issues/40
+use ashpd::WindowIdentifier;
 
 use log::error;
 use pollster::block_on;
+use raw_window_handle::{RawDisplayHandle, RawWindowHandle};
+
+fn to_window_identifier(
+    window: Option<RawWindowHandle>,
+    display: Option<RawDisplayHandle>,
+) -> Option<WindowIdentifier> {
+    window.map(|window| {
+        block_on(Box::pin(async move {
+            WindowIdentifier::from_raw_handle(&window, display.as_ref()).await
+        }))
+    })
+}
```

Comment:
Could we drop the block_on and make `to_window_identifier` async instead?
Every call-site of `to_window_identifier` is in async context already anyway.
```suggestion
async fn to_window_identifier(
    window: Option<RawWindowHandle>,
    display: Option<RawDisplayHandle>,
) -> Option<WindowIdentifier> {
    let identifier = WindowIdentifier::from_raw_handle(&window?, display.as_ref()).await;
    Some(identifier)
}
```